tests/sles4sap/publiccloud/qesap_terraform.pm

```diff
@@ -171,12 +167,22 @@ sub run {
     die "HA cluster needs at least 2 nodes. Check 'NODE_COUNT' parameter." if ($ha_enabled && (get_var('NODE_COUNT') <= 1));
 
     set_var('FENCING_MECHANISM', 'native') unless ($ha_enabled);
-    if (is_azure) {
-        # Update PUBLIC_CLOUD_RESOURCE_GROUP in Azure's tests so it includes the random
-        # string appended to the workspace
-        set_var('PUBLIC_CLOUD_RESOURCE_GROUP', $workspace);
-        record_info 'Resource Group', "Resource Group used for deployment: $workspace";
-    }
+
+    my $deployment_name = qesap_calculate_deployment_name(get_var('PUBLIC_CLOUD_RESOURCE_GROUP', 'qesaposd'));
+    # Create a QESAP_DEPLOYMENT_NAME variable so it includes the random
+    # string appended to the PUBLIC_CLOUD_RESOURCE_GROUP
+    #
+    # User can :
+    #  * define none of PUBLIC_CLOUD_RESOURCE_GROUP and QESAP_DEPLOYMENT_NAME
+    #     resulting deployment_name: qesaposd123456
+    #  * only define PUBLIC_CLOUD_RESOURCE_GROUP=goofy
+    #     resulting deployment_name: goofy123456
+    #  * define QESAP_DEPLOYMENT_NAME=goofy
+    #     resulting deployment_name: goofy
+    #     PUBLIC_CLOUD_RESOURCE_GROUP is completly ignored and
+    #     the job_id is not included in the deployment name
+    set_var('QESAP_DEPLOYMENT_NAME', get_var('QESAP_DEPLOYMENT_NAME', $deployment_name));
+    record_info 'Resource Group', "Resource Group used for deployment: $deployment_name";
```

Comment:
It could be better to add description of this new QESAP_DEPLOYMENT_NAME to variables.md also.